src/EnergyPlus/HVACManager.cc

```diff
@@ -3005,6 +3009,30 @@ namespace HVACManager {
         }
     }
 
+    void CheckAirLoopFlowBalance()
+    {
+        // Check for unbalanced airloop
+        if (!isPulseZoneSizing && !DataHeatBalance::ZoneAirMassFlow.EnforceZoneMassBalance && !DataGlobals::WarmupFlag && AirLoopsSimOnce) {
```

Comment:
This is out of your scope but why don't we care about a mass imbalance if we are doing pulse sizing or enforcing mass balance?

Reply:
One other comment, put an outside IF block using IF(thisAirLoopFlow.FlowError) so these multiple checks don't have to be evaluated each call.